src/ContentScript/observer.js

```diff
@@ -0,0 +1,35 @@
+/** @type {MutationCallback}
+ * @description make and observer callback function to process new nodes
+ */
+export function makeObserverCallback(parserFn) {
+  function observeNewlyAddedNodesCallback(/** @type {MutationrRecord[]} */ mutationRecords) {
+    mutationRecords.forEach(({ type, addedNodes }) => {
+      if (type !== 'childList') return;
+
+      addedNodes?.forEach(parserFn);
+    });
+  }
+
+  return observeNewlyAddedNodesCallback;
+}
+
+/**
+ * @description setup an observer on a target node
+ */
+export function runObserver(observer, target, observerCallback) {
```

Comment:
it would be nice to group these functions and abstract the observer to a class, we can do this since the code will be polyfilled

Reply:
i not sure i follow when you say the observer will be polyfilled, but i have gone ahead and implemented rewritten those functions into a class. As well as made a logger class for dealing with development runtime coding where i need to log errors and what not